### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by alexandru-dinu** [alexandru-dinu/3D-face-reconstruction#2]
- [x] aggregate model summaries (**HG**, U-Net, VGG, VRN-Unguided, FAN)
- [x] visualize model output in meshlab via octave
    - [x] **why is it so sparse and long on z?**
- [ ] **use a feature extractor trained on faces**
    - [ ] **also try with their FAN** 
---

- [x] **landmark heatmaps**
    - [x] check if landmarks from dataset are aligned with the volume
- [x] double check u-net <-> hourglass arch
    - [x]  implement standalone hourglass module
- [x] model output visualization
    - [x] construct denser representation
    - [ ] **marching-cubes**
- [x] overfit smaller dataset
- [x] write dataloader (3D facial scan -> 3D binary volume)
- [ ] generate 3D coords from .mat (where not present)
- [x] convert torch model to pytorch and visualize it
- [x] data augmentation (transformations on labels)

---

### Tried
- smaller u-net + stacked u-net
- bigger 2xHourglass
- predict depth-map instead of volume
    - feature extraction (VGG) + 2xFC (relu + dropout)
    - additional loss term
- closing + gaussian smoothing
- vrn_unguided from torch (with `UpsamplingBilinear2d` / `ConvTranspose2d`)
- weighted loss on gaussian landmarks
